docs/topics/whatsnew-eap.md

```diff
@@ -269,9 +562,173 @@ There are new Gradle properties so that you can configure a directory of your ch
+* [Support for type-safe plain JavaScript objects](#support-for-type-safe-plain-javascript-objects)
+* [Support for npm package manager](#support-for-npm-package-manager)
+ 
+### Support for type-safe plain JavaScript objects
+
+> The `js-plain-objects` plugin is [Experimental](components-stability.md#stability-levels-explained).
+> It may be dropped or changed at any time. The `js-plain-objects` plugin **only** supports the K2 compiler.
+>
+{type="warning"}
+
+To make it easier to work with JavaScript APIs, in Kotlin 2.0.0-Beta4 we provide a new plugin: 
+[`js-plain-objects`](https://github.com/JetBrains/kotlin/tree/master/plugins/js-plain-objects),
+that you can use to create type-safe plain JavaScript objects.
+The plugin checks your code for any [external interfaces](wasm-js-interop.md#external-interfaces) 
+that have a `@JsPlainObject` annotation and adds:
+
+* an inline `invoke` operator function inside the companion object that you can use as a constructor.
+* a `.copy()` function that you can use to create a copy of your object while adjusting some of its properties.
+
+For example:
+
+```kotlin
+import kotlinx.js.JsPlainObject
+
+@JsPlainObject
+external interface User {
+    var name: String
+    val age: Int
+    val email: String? 
+}
+
+fun main() {
+    // Creates a JavaScript object
+    val user = User(name = "Name", age = 10)
+    // Copies the object and adds an email
+    val copy = user.copy(age = 11, email = "some@user.com")
+
+    println(JSON.stringify(user)) 
+    // { "name": "Name", "age": 10 }
+    println(JSON.stringify(copy)) 
+    // { "name": "Name", "age": 11, "email": "some@user.com" }
+}
+```
+
+Any JavaScript objects created with this approach are safer because instead of only seeing errors at runtime, 
+you can see them at compile time, or even highlighted by your IDE.
+
+Consider this example that uses a `fetch()` function to interact with a JavaScript API, using external interfaces 
+to describe the shape of the JavaScript objects:
+
+```kotlin
+import kotlinx.js.JsPlainObject
+
+@JsPlainObject
+external interface FetchOptions {
+    val body: String?
+    val method: String
+}
+
+// A wrapper for Window.fetch
+suspend fun fetch(url: String, options: FetchOptions? = null) = TODO("Add your custom behavior here")
+
+// A compile-time error is triggered as metod is not recognized
+fetch("https://google.com", options = FetchOptions(metod = "POST")) 
+// A compile-time error is triggered as method is required
+fetch("https://google.com", options = FetchOptions(body = "SOME STRING")) 
+```
+
+In comparison, if you use the `js()` function instead to create your JavaScript objects,
+errors are only found at runtime or aren't triggered at all:
+
+```kotlin
+suspend fun fetch(url: String, options: FetchOptions? = null) = TODO("Add your custom behavior here")
+
+// No error is triggered. As metod is not recognized, the wrong method 
+// (GET) is used.
+fetch("https://google.com", options = js("{ metod: 'POST' }")) 
```

Comment:
```suggestion
fetch("https://google.com", options = js("{ method: 'POST' }")) 
```

Reply:
This would break the example so I'm not including it.